cmd/kube-apiserver/app/aggregator.go

```diff
@@ -132,7 +133,8 @@ func createAggregatorServer(aggregatorConfig *aggregatorapiserver.Config, delega
 	apiServices := apiServicesToRegister(delegateAPIServer, autoRegistrationController)
 	crdRegistrationController := crdregistration.NewAutoRegistrationController(
 		apiExtensionInformers.Apiextensions().InternalVersion().CustomResourceDefinitions(),
-		autoRegistrationController)
+		autoRegistrationController,
+		crdOpenAPIAggregationManager)
```

Comment:
is this only used for CRDs?

Reply:
Yes. `crdOpenAPIAggregationManager` holds the openapi services of apiextensions apiserver. https://github.com/kubernetes/kubernetes/pull/67205/files#diff-36e1d7088209b1ab6ff878b18bb35210R200